R/tar_forecast_summaries.R

```diff
@@ -0,0 +1,96 @@
+
+#' ecmwf_summarise_seasons
+#' @details
+#' Works for default windows of "primera" and "postera" and Copernicus Data Store Public SEAS51 data
+#' **Note** when we get access to HRES, we will need to adjust line that creates `monitoring_mo_seq` as will have an extra
+#' month of lead time data
+#' **Note** -- sequencing of months would be disrupted/faulty if we start using months that extend into next year i.e (11,12,1,2,)
+#' 
+#' @param df data.frame with all ECMWF data aggregated to monthly precip values for each leadtime
+#' @param window_list list containing numeric vector representing months
+
+summarise_seasons <-  function(df=df_ecmwf_zonal_all,
+                                    window_list= list("primera"=c(5,6,7,8),
+                                                      "postera"=c(9,10,11)),
+                                    forecast_source ="mars"
+                                                      ){
+  df <- df %>% 
```

Comment:
yeah good question - let me see if I can find a clear answer on this.

I know it's standard to define the packages needed for your targets with `tar_option_set()` and let this control all package handling for the pipeline - i think it does some slick stuff under the hood ( [I should read this more closely](https://github.com/ropensci/targets/discussions/849) ).

this being said, I don't think it's an issue to have the the `library()` calls at the top of R-script. Usually I see the recommended to just use `::` in functions, but agreed with you in a previous conversation that this makes the code less legible. Is this still your thinking or has that changed with adoption of `box` where you are just using `$`-> instead of `::` ?

Reply:
cool - added/cleaned up library calls with [8a7c289](https://github.com/OCHA-DAP/ds-aa-lac-dry-corridor/pull/12/commits/8a7c289dce2b216282f318e9609668c3e172b061)